app/apollo/models/channel.schema.js

```diff
@@ -46,6 +46,12 @@ const ChannelSchema = new mongoose.Schema({
     type: Date,
     default: Date.now,
   },
+  custom: [
+    {
+      key: String,
+      val: String,
+    }
+  ],
```

Comment:
With apologies, I know I suggested the array, but ....

What happens if a user specifies the same key more than once. Would this work/work better if this was an object?  It would prevent duplicate keys, but with what other implications?